[SOL] lldb: Add solana-lldb wrapper

jawilk opened the pull request:
Moves https://github.com/solana-labs/rust/pull/67 into `lldb/startup/solana`

This adds a wrapper around the LLDB executable. Once we start to distribute LLDB in sbf-tools, the wrapper + scripts can be put into the same directory as the LLDB executable. Invoking `./solana-lldb` will pre-load helper functions to print rust as well as solana types (e.g. pubkeys in base58 or auto deref of accounts) and then start the LLDB executable.

The original rust scripts are also added in this PR and taken from https://github.com/solana-labs/rust/tree/solana-1.62.0/src/etc as is.

dmakarov commented:
Wouldn't it be better to place these scripts under lldb/scripts/solana (instead of lldb/startup/solana)? What do you think?

jawilk committed:
[SOL] lldb: Add solana-lldb wrapper

This adds a wrapper around the LLDB executable that resides in the same directory. Once we start to distribute LLDB in sbf-tools, the wrapper + scripts can be put into the same directory as the LLDB executable. Invoking ./solana-lldb will pre-load helper functions to print rust as well as solana types (e.g. pubkeys in base58 or auto deref of accounts) and then start the LLDB executable.

jawilk commented on lldb/scripts/solana/solana-lldb:
```diff
@@ -0,0 +1,27 @@
+
+set -e
+
+lldb=lldb
```
should be `lldb=./lldb`
